pkg/common/file.go

```diff
@@ -34,175 +35,123 @@ type Config struct {
 	Debug             bool
 	PullImageOnCreate bool
 	DisablePullOnRun  bool
-	yamlData          *yaml.Node // YAML representation of config
 }
 
+const (
+	runtimeEndpointKey   = "runtime-endpoint"
+	imageEndpointKey     = "image-endpoint"
+	timeoutKey           = "timeout"
+	debugKey             = "debug"
+	pullImageOnCreateKey = "pull-image-on-create"
+	disablePullOnRunKey  = "disable-pull-on-run"
+)
+
 // ReadConfig reads from a file with the given name and returns a config or
 // an error if the file was unable to be parsed.
-func ReadConfig(filepath string) (*Config, error) {
-	data, err := os.ReadFile(filepath)
+func ReadConfig(path string) (*Config, error) {
```

Comment:
maybe easier to review if there are some unit tests for this function.

Reply:
I agree, we should add them before this PR.

/hold

Reply:
Also those edge cases: https://github.com/kubernetes-sigs/cri-tools/pull/1646/files#r1811995991

I am curious if parsing to map would have troubles with the duplicate entries and whether it will keep which value will be used.

Reply:
Rebased the PR and fixed the logic accordingly to the failing test.s

Reply:
thank you!